core/src/toga/widgets/selection.py

```diff
@@ -1,100 +1,155 @@
+from __future__ import annotations
+
 import warnings
 
 from toga.handlers import wrapped_handler
+from toga.sources import ListSource
 
 from .base import Widget
 
 
 class Selection(Widget):
-    """The Selection widget lets you pick from a defined selection of options.
-
-    Args:
-        id (str): An identifier for this widget.
-        style ( :obj:`Style`): An optional style object.
-            If no style is provided then a new one will be created for the widget.
-        items (``list`` of ``str``): The items for the selection.
-    """
-
     def __init__(
         self,
         id=None,
         style=None,
-        items=None,
-        on_select=None,
+        items: list | ListSource | None = None,
+        accessor: str | None = None,
+        value: None = None,
+        on_change: callable | None = None,
         enabled=True,
-        factory=None,  # DEPRECATED!
+        on_select: callable | None = None,  # DEPRECATED
     ):
+        """Create a new Slider widget.
+
+        Inherits from :class:`~toga.widgets.base.Widget`.
+
+        :param id: The ID for the widget.
+        :param style: A style object. If no style is provided, a default style
+            will be applied to the widget.
+        :param items: The items to display for selection. Can be either a list
+            of literal items, or a ListSource.
+        :param accessor: The accessor for the ListSource. If ``items`` is
+            provided as a list of items that are non-string iterable objects,
+            the ``accessor`` is used to select the attribute of the item that
+            will be used for display purposes. Not required if the ``items`` is
+            a list of strings, a list of non-iterable objects, or a ListSource.
```

Comment:
I find this very confusing, because there's no obvious relationship between an object being iterable, and accessing one of its attributes rather than displaying the object itself. Neither does this obviously cover the case of a list of dicts, as shown in the example code.

The similar text at `items` below has the same problem.

Reply:
Agreed this is confusing; I've tried a second draft.